blog/02-09-2022-creating-a-provider-for-the-go-sdk.mdx

```diff
@@ -0,0 +1,383 @@
+
+## Providers
+
+A [Provider](https://docs.openfeature.dev/docs/specification/sections/providers) is responsible for performing flag evaluation, they can be as simple as a local JSON evaluator, or act as an abstraction layer for a more complex evaluation system. Only one `Provider` can be registered at a time, and OpenFeature will no-op if one has not been defined. Before writing your own `Provider`, it is strongly recommended to familiarize yourself with the [OpenFeature spec](https://docs.openfeature.dev/docs/specification/).  
+In this tutorial I will demonstrate the steps taken to create a new `Provider` whilst conforming to the OpenFeature spec using a simple flag implementation. The flag evaluation will be handled by a simple JSON evaluator and flag configurations will be stored as environment variables.  
+
+
+The following section describes how the flag evaluator portion of this project has been constructed, to skip to the `Provider` specific implementations, [click here](#creating-a-compliant-provider).  
+
+## Creating the flag evaluator
+
+The core of any flag `Provider` is the evaluation logic used to provide the flag values from the provided metadata (referred to as the [Evaluation Context](https://docs.openfeature.dev/docs/specification/sections/evaluation-context)). For this example I have put together a very simple json evaluator. 
+Flags are configured using the structs described below, and are stored as environment variables:
+```go
+type StoredFlag struct {
+	DefaultVariant string    `json:"defaultVariant"`
+	Variants       []Variant `json:"variant"`
+}
+
+type Variant struct {
+	Criteria     []Criteria  `json:"criteria"`
+	TargetingKey string      `json:"targetingKey"`
+	Value        interface{} `json:"value"`
+	Name         string      `json:"name"`
+}
+
+type Criteria struct {
+	Key   string      `json:"key"`
+	Value interface{} `json:"value"`
+}
+```
+example JSON:
+```json
+{
+    "defaultVariant":"not-yellow",
+    "variants": [
+        {
+			"name": "yellow-with-key",
+            "targetingKey":"user",
+            "criteria": [
+                {
+                    "color":"yellow"
+                }
+            ],
+            "value":true
+        },
+		{
+			"name": "yellow",
+            "targetingKey":"",
+            "criteria": [
+                {
+                    "color":"yellow"
+                }
+            ],
+            "value":true
+        },
+        {
+			"name": "not-yellow",
+            "targetingKey":"",
+            "criteria": [],
+            "value":false
+        }
+	]
+}
+```
+Each flag value contains an array of `Variants`, each with their own array of `Criteria`. When a flag request needs to be evaluated, the `Variants` slice is iterated over, if the `EvaluationContext` matches all required `Criteria` for a specific `Variant`, the associated flag value will be returned from the evaluator. If a matching `Variant` is not found the `DefaultVariant` is returned in the response.
```

Comment:
You may want to link to the definitions for variants: https://docs.openfeature.dev/docs/specification/glossary#variant

Reply:
I have added this link